include/pybind11/detail/common.h

```diff
@@ -1031,8 +1028,12 @@ PYBIND11_NAMESPACE_END(detail)
 ///  - regular: static_cast<Return (Class::*)(Arg0, Arg1, Arg2)>(&Class::func)
 ///  - sweet:   overload_cast<Arg0, Arg1, Arg2>(&Class::func)
 template <typename... Args>
+#    if (defined(_MSC_VER) && _MSC_VER < 1920) /* MSVC 2017 */                                    \
+        || (defined(__clang__) && __clang_major__ == 5)
 static constexpr detail::overload_cast_impl<Args...> overload_cast = {};
-// MSVC 2015 only accepts this particular initialization syntax for this variable template.
+#    else
+static constexpr detail::overload_cast_impl<Args...> overload_cast;
+#    endif
```

Comment:
Why is the workaround longer now?

Reply:
Before there was **only** the workaround, with a comment that marked it as such.

Now there is what (I assume) the original authors wanted to use, but couldn't, plus the workaround.

I could change it back to only-workaround-with-updated-comment, but at this point in time we're just a couple years away from being able to purge the workaround. I think it's nice to be sure that there aren't any modern compilers having trouble with the default construction. It's kind-of the point of beta testing compilers, that we want to catch such things.